core/src/main/java/org/apache/druid/utils/CollectionUtils.java

```diff
@@ -68,6 +71,17 @@ public int size()
     };
   }
 
+  /**
+   * Returns a transformed map from the given input map where the value is modified based on the given valueMapper
+   * function.
+   */
+  public static <K, V, V2> Map<K, V2> transformValues(Map<K, V> map, Function<V, V2> valueMapper)
+  {
+    final Map<K, V2> result = new HashMap<>();
```

Comment:
Better to use `Maps.newHashMapWithExpectedSize(map.size())` here

Reply:
sure